app/src/main/java/org/mozilla/fenix/experiments/NimbusSetup.kt

```diff
@@ -88,32 +95,24 @@ fun createNimbus(context: Context, url: String?): NimbusApi {
             // generated code.
             register(observer)
 
-            // This performs the minimal amount of work required to load branch and enrolment data
-            // into memory. If `getExperimentBranch` is called from another thread between here
-            // and the next nimbus disk write (setting `globalUserParticipation` or
-            // `applyPendingExperiments()`) then this has you covered.
-            // This call does its work on the db thread.
-            initialize()
+            val isFirstNimbusRun = context.settings().isFirstNimbusRun || url.isNullOrBlank()
 
-            if (!enabled) {
-                // This opts out of nimbus experiments. It involves writing to disk, so does its
-                // work on the db thread.
-                globalUserParticipation = enabled
+            runBlocking {
+                initialize(
+                    isFirstNimbusRun,
+                    R.raw.initial_experiments,
+                ).joinOrTimeout(TIME_0UT_LOADING_EXPERIMENT_FROM_DISK_MS)
             }
 
-            if (context.settings().isFirstNimbusRun || url.isNullOrBlank()) {
-                setExperimentsLocally(R.raw.initial_experiments)
+            if (isFirstNimbusRun) {
                 context.settings().isFirstNimbusRun = false
             }
 
-            // We may have downloaded experiments on a previous run, so let's start using them
-            // now. We didn't do this earlier, so as to make getExperimentBranch and friends returns
-            // the same thing throughout the session. This call does its work on the db thread.
-            applyPendingExperiments()
-
-            // Now fetch the experiments from the server. These will be available for feature
-            // configuration on the next run of the app. This call launches on the fetch thread.
-            fetchExperiments()
+            if (!enabled) {
+                // This opts out of nimbus experiments. It involves writing to disk, so does its
+                // work on the db thread.
+                globalUserParticipation = enabled
+            }
```

Comment:
I think we can delete this now (it was there to migrate from mako).

```suggestion
```

Reply:
Do we need to pass `enabled` to nimbus to know when users opt-out of experiments on the Fenix ui?

Reply:
Maybe we should address this in separate ticket, as I'm not sure if this is something we would like to uplift to v106.

Reply:
Filed https://github.com/mozilla-mobile/fenix/issues/27110.

Reply:
ty!